packages/xstate-graph/src/types.ts

```diff
@@ -187,10 +183,7 @@ export interface TraversalConfig<TState, TEvent extends EventObject>
    * `event` when building the adjacency map.
    */
   filter: (state: TState, event: TEvent) => boolean;
-  getEvents: (
-    state: TState,
-    cases: EventCaseMap<TState, TEvent>
-  ) => ReadonlyArray<TEvent>;
+  events: TEvent[] | ((state: TState) => TEvent[]);
```

Comment:
```suggestion
  events: readonly TEvent[] | ((state: TState) => readonly TEvent[]);
```

this change should come with 2 tests that would exercise `as const` on the whole array (and not on the array elements)

Reply:
This doesn't work:

```ts
  it('`events` should not require all event types (function)', () => {
    const machine = createMachine<
      unknown,
      { type: 'FOO'; value: number } | { type: 'BAR'; value: number }
    >({});

    getMachineShortestPaths(machine, {
      events: () => [{ type: 'FOO', value: 100 }] as const
    });
  });
```

But this does:
```ts
  it('`events` should not require all event types (function)', () => {
    const machine = createMachine<
      unknown,
      { type: 'FOO'; value: number } | { type: 'BAR'; value: number }
    >({});

    getMachineShortestPaths(machine, {
      events: () => [{ type: 'FOO', value: 100 } as const]
    });
  });
```

Reply:
I think that you didn't rebuild the project or something. I had the same problem but after running `yarn build` it went away.

I pushed out this change and the tests, let's wait for the CI to be extra sure